xslices/xslices.go

```diff
@@ -140,3 +140,21 @@ func CopyN[S ~[]V, V any](s S, n int) S {
 
 	return result
 }
+
+// Deduplicate removes duplicate elements from the slice based on the key function provided.
+func Deduplicate[T any, K comparable](items []T, keyFunc func(T) K) []T {
+	var (
+		result []T
+		seen   = make(map[K]bool)
```

Comment:
can simply use no value, no?
```suggestion
		seen   = make(map[K]struct{})
```

Reply:
Ah yes! Fixed.